### PR TITLE
OCPBUGS-11439: allow z-stream upgrade even if CVO Upgradeable is false

### PR DESCRIPTION
**Pull request by sjenning**
https://issues.redhat.com/browse/HOSTEDCP-632

Refactored version of https://github.com/openshift/hypershift/pull/2336 to address https://github.com/openshift/hypershift/pull/2336#discussion_r1151394058 and fixup/add unit tests

fyi @wking

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -4180,22 +4185,54 @@ func (r *HostedClusterReconciler) reconcileAWSSubnets(ctx context.Context, creat
...
+	}
+	return r.ReleaseProvider.Lookup(ctx, hcluster.Spec.Release.Image, pullSecretBytes)
+}
+
+// isUpgradeable returns
+// 1) bool indicating whether the HostedCluster is upgrading
+// 2) non-error message about the condition of the upgrade
+// 3) error indicating that the upgrade is not allowed or we were not able to determine
+func isUpgradeable(hcluster *hyperv1.HostedCluster, releaseImage *releaseinfo.ReleaseImage) (bool, string, error) {
+	if hcluster.Status.Version == nil || hcluster.Status.Version.Desired.Image == hcluster.Spec.Release.Image {
```
Comment (sjenning) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:4205]: Reviewer note: I inverted some of the logic here so we can just return early and make it clearer.

**Comment by celebdor** [comment]
/retitle OCPBUGS-11439: allow z-stream upgrade even if CVO Upgradeable is false

**Pull request merged**